### PR TITLE
feat(MWPW-142267): Merch What's Included and Merch Mnemonic List (TwP)

### PR DESCRIPTION
**Pull request by Axelcureno**
Introduces Merch What's Included and Merch Mnemonic List web components for TwP What's included screen.
Resolves: MWPW-142267
Test link: https://mwpw-142267--milo--adobecom.hlx.page/drafts/axel/block-samples/twp/illustrator#twpmodal
Migrated from https://git.corp.adobe.com/wcms/tacocat.js/pull/572
d8d35034-689b-47b0-8079-a935ec774a8a

How it works in Milo
In merch-twp block
Given a merch-twp block, a third separator --- is expected and then below a link to the what's included screen modal content.
fef4cdb1-2418-4218-b6d5-50307bbbf348

In merch-card block
A link with name "merch-whats-included" is expected (link text is authorable)
e56673f8-7c12-4a54-a1e1-0872c7a81fed

This PR also introduces 2 new web components, which are:

merch-whats-included
82c91645-1ef5-42e1-8034-79eaeac77f59
Contains link of whats-included modal content (works as an inline fragment) and lets the author configure number of rows in mobile to display before "+ See more".

merch-mnemonic-list
c7ccda5c-6ccf-458f-9392-fd4eb1b5648e

Contains content of what's included screen.

Assumptions
What's included is an experience only applicable to a CC All Apps merch card.
Supports unlimited number of columns
Supports configurable number of rows before displaying "+ See more" in mobile.

Preview URLs:
- Before: https://main--milo--adobecom.hlx.page/drafts/axel/block-samples/twp/illustrator#twpmodal
- After: https://mwpw-142267--milo--adobecom.hlx.page/drafts/axel/block-samples/twp/illustrator#twpmodal

URL for testing:
- https://mwpw-142267--mas--adobecom.hlx.page/?martech=off

### CODE REVIEW

File: web-components/test/merch-whats-included.test.html [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+                    </merch-mnemonic-list>
+                    <merch-mnemonic-list>
+                        <div slot="icon">
+                            <merch-icon size="s" src="https://www.adobe.com/content/dam/shared/images/product-icons/svg/illustrator.svg" alt="Illustrator"></merch-icon>
+                        </div>
+                        <p slot="description"><strong>Illustrator</strong></p>
+                    </merch-mnemonic-list>
+                    <merch-mnemonic-list>
+                        <div slot="icon">
+                            <merch-icon size="s" src="https://www.adobe.com/content/dam/shared/images/product-icons/svg/premiere.svg" alt="Premiere Pro"></merch-icon>
```
[the remark this answers is not in the document]
Comment (Axelcureno) [reply web-components/test/merch-whats-included.test.html]: Fixed

**Comment by Axelcureno** [comment]
> Overal I like your approach with `[slot="merch-whats-included"]`
> 
> 1. There are some visual issues on mobile.
> 
> image
> 
> 2. expecting a fragment with a specific name feels a bit limiting to me. Couldn't we use a `#merch-whats-included`  hash or something similar?
> 3. How to we close and go back to TWP without closing the modal?

Please find my answers below.
1. Fixed.
2: Using a hash will make Milo treat the content as Modal which complicates things and would require a "hacky" approach to not update the URL and capture the click event of such modal. Using query parameters also doesn't work with fragments since those are stripped by milo. In my example it is not specifically tied to that name, it could be a different name as long as the name contains 'merch-whats-included'. Another thing to keep in mind is that the end user will not see this file name at all (unless inspecting the code). Please suggest.

3. By capturing the close modal event and toggling if the fragment content visibility. I have started a [thread](https://adobe-mwp.slack.com/archives/C03HAU5SU2E/p1720482008154509) in #milo-dev

**Review by 3ch023 — APPROVED** [review]

**Comment by 3ch023** [comment]
@Axelcureno to make the PSI check green you can just provide a fake link, with your branch, you can check the example green pr here:
https://github.com/adobecom/mas/pull/19

**Comment by Axelcureno** [comment]
URL for testing:

- https://mwpw-142267--mas--adobecom.hlx.page/

**Comment by Axelcureno** [comment]
> Introduces Merch What's Included and Merch Mnemonic List web components for TwP What's included screen. Resolves: MWPW-142267 Test link: https://mwpw-142267--milo--adobecom.hlx.page/drafts/axel/block-samples/twp/illustrator#twpmodal Migrated from https://git.corp.adobe.com/wcms/tacocat.js/pull/572 d8d35034-689b-47b0-8079-a935ec774a8a
> 
> How it works in Milo In merch-twp block Given a merch-twp block, a third separator --- is expected and then below a link to the what's included screen modal content. fef4cdb1-2418-4218-b6d5-50307bbbf348
> 
> In merch-card block A link with name "merch-whats-included" is expected (link text is authorable) e56673f8-7c12-4a54-a1e1-0872c7a81fed
> 
> This PR also introduces 2 new web components, which are:
> 
> merch-whats-included 82c91645-1ef5-42e1-8034-79eaeac77f59 Contains link of whats-included modal content (works as an inline fragment) and lets the author configure number of rows in mobile to display before "+ See more".
> 
> merch-mnemonic-list c7ccda5c-6ccf-458f-9392-fd4eb1b5648e
> 
> Contains content of what's included screen.
> 
> Assumptions What's included is an experience only applicable to a CC All Apps merch card. Supports unlimited number of columns Supports configurable number of rows before displaying "+ See more" in mobile.
> 
> Preview URLs:
> 
> * Before: https://main--milo--adobecom.hlx.page/drafts/axel/block-samples/twp/illustrator#twpmodal
> * After: https://mwpw-142267--milo--adobecom.hlx.page/drafts/axel/block-samples/twp/illustrator#twpmodal
> 
> URL for testing:
> 
> * https://mwpw-142267--mas--adobecom.hlx.page/?martech=off

File: web-components/build.mjs [hunk 2]
```diff
@@ -92,11 +92,41 @@ Promise.all([
         plugins: [rewriteImports()],
         external: ['lit'],
     }),
+    build({
+        entryPoints: ['./src/merch-twp-d2p.js'],
+        bundle: true,
+        banner,
+        outfile: './lib/merch-twp-d2p.js',
```
Comment (yesil) [inline web-components/build.mjs:99]: merch-twp-d2p is already being build below.
also ./lib has changed to ./libs

File: web-components/build.mjs [hunk 3]
```diff
@@ -92,11 +92,41 @@ Promise.all([
...
+        external: ['lit'],
+    }),
+    build({
+        entryPoints: ['./src/merch-mnemonic-list.js'],
+        bundle: true,
+        banner,
+        outfile: './lib/merch-mnemonic-list.js',
+        format: 'esm',
+        plugins: [rewriteImports()],
+        external: ['lit'],
```
Comment (yesil) [inline web-components/build.mjs:120]: I think these build entries are not needed.

**Review by yesil — COMMENTED** [review]
let's merge for now, but prevent GWP from authoring yet with this

**Review by yesil — APPROVED** [review]

**Pull request merged**